src/main/java/uk/gov/justice/hmpps/datacompliance/repository/jpa/repository/ual/OffenderUalRepository.java

```diff
@@ -1,19 +1,21 @@
 package uk.gov.justice.hmpps.datacompliance.repository.jpa.repository.ual;
 
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import uk.gov.justice.hmpps.datacompliance.repository.jpa.model.ual.OffenderUalEntity;
 
-import java.util.List;
 import java.util.Optional;
 
 
 @Repository
 public interface OffenderUalRepository extends CrudRepository<OffenderUalEntity, Long> {
 
-    Optional<OffenderUalEntity> findOneByOffenderNo(final String offenderNo);
+    Optional<OffenderUalEntity> findOneByOffenderNoIgnoreCase(final String offenderNo);
 
-    Optional<OffenderUalEntity> findOneByOffenderBookingNoAndOffenderCroPncAndFirstNamesAndLastName(final String offenderBookingNo, final String offenderCroPnc, final String firstNames, final String lastName);
+    Optional<OffenderUalEntity> findOneByOffenderBookingNoIgnoreCase(final String offenderBookingNo);
 
-    void deleteByOffenderUalIdNotIn(final List<Long> ids);
+    Optional<OffenderUalEntity> findOneByOffenderPncIgnoreCase(@Param("pnc") final String pnc);
+
+    Optional<OffenderUalEntity> findOneByOffenderCroIgnoreCase(@Param("cro") final String cro);
```

Comment:
I don't really understand what @Param does with a query?

Reply:
This is from the reminisce of a work around for a hibernate bug from when I was using contains. You can read about it [here](https://github.com/spring-projects/spring-data-jpa/issues/2472). In any case it is no longer required so I have removed it.

Reply:
Ha! We bumped into a very similar issue recently with that escape parameter in a generated query - [had to replace with a custom query](https://github.com/ministryofjustice/send-legal-mail-to-prisons-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactRepository.kt#L21).